ogc-wfs/src/main/java/io/arlas/server/ogc/wfs/filter/FilterToElastic.java

```diff
@@ -1015,8 +1017,41 @@ protected Object evaluateLiteral(Literal expression, Class<?> target) {
      * @param literal
      */
     protected void writeLiteral(Object literal) {
+        boolean isDate = false;
+        if(((String) field).split(":").length>1){
+            String[] pathElements = ((String) field).split(":")[1].split("_");
+            isDate = isPathDate(pathElements,collectionReference.properties);
+        }else if(((String) field).split(":").length==1){
+            String[] pathElements = ((String) field).split("_");
+            isDate = isPathDate(pathElements,collectionReference.properties);
+        }
         field = literal;
-        dateFormatter = DEFAULT_DATE_FORMATTER;
+        if(isDate && !Date.class.isAssignableFrom(literal.getClass())){
+            SimpleDateFormat f = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXX");
+            f.setTimeZone(TimeZone.getTimeZone("UTC"));
+            try {
+                //hack for the ets test suite wich test with bad date
+                // bad 1970-01-01T00:13:33.4Z
+                // bad 1970-01-01T00:13:33.43Z
+                // good 1970-01-01T00:13:33.430Z
+                String lastElement = ((String)literal).split("\\.")[1];
+                String firstElement = ((String)literal).split("\\.")[0];
+                String millisPart = lastElement.trim();
+                if(millisPart.length()==1){
+                    literal =firstElement.concat(".").concat(millisPart).concat("00").concat("Z");
+                }else if(lastElement.trim().length()==2){
+                    literal =firstElement.concat(".").concat(millisPart).concat("0").concat("Z");
+                }
+                //TO DO change the test
+                field = dateFormatter.print((f.parse((String)literal)).getTime());
+            } catch (ParseException e) {
+                List<OGCExceptionMessage> wfsExceptionMessages = new ArrayList<>();
+                wfsExceptionMessages.add(new OGCExceptionMessage(OGCExceptionCode.OPERATION_PROCESSING_FAILED, "Invalid Filter", "filter"));
+                wfsExceptionMessages.add(new OGCExceptionMessage(OGCExceptionCode.INVALID_PARAMETER_VALUE, "Unable to find " + field + "  in " + collectionReference.collectionName + ".", "filter"));
```

Comment:
I think we should put `"Unable to format " + field` for example in exception message